### PR TITLE
Slack attachment in email notifications

### PR DESCRIPTION
**Pull request by harshilsharma63**
#### Summary
Included post slack attachments in email notifications. The implementation is agnostic of who generated that post, be it Focalboard plugin or the CircleCI plugin, webhook notification, or anything.

#### Ticket Link
https://community.mattermost.com/boards/team/qghzt68dq7bopgqamcnziq69ao/b4ctmm917q3rzdfi451qtuxjzcw/392c8889-6c8f-4cb7-8407-f5c57c1acfe8/cpcb81equzpdobpeyn8fqptkgmc

#### Release Note
```release-note
post email notifications now include `SlackAttachments` from the post as well.
```

#### Screenshots
Screenshot 2022-09-14 at 4 08 09 PM
Screenshot 2022-09-14 at 4 08 51 PM
Screenshot 2022-09-14 at 4 53 34 PM

### CODE REVIEW

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:  Great work @harshilsharma63 !

**Comment by harshilsharma63** [comment]
@hmhealey could you please review this next week?

File: templates/messages_notification.html [hunk 1]
```diff
@@ -350,7 +350,156 @@
         padding: 0px 0px 0px 12px !important;
       }
     }
+
+    .slackAttachmentContent {
```
Comment (hmhealey) [inline templates/messages_notification.html:354]: Would you mind renaming these `messageAttachmentX`? We've moved everything outside the server code to refer to these as "message attachments", and while renaming it everywhere is blocked by being a breaking change for people using the Go API, I think we probably want to use the new name in new code

**Review by hmhealey — COMMENTED** [review]
Quick comment since I noticed something I wanted to mention, but Harshil and I discussed having someone else review this since I'm not particularly familiar with server code at the moment

File: templates/messages_notification.html (hunk 1 above)
Comment (harshilsharma63) [reply templates/messages_notification.html]: Done!

Comment (hmhealey) [reply templates/messages_notification.html]: Thanks!

File: templates/partials/sender_info_col.mjml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<mj-column css-class="senderInfoCol" width=90% >
+    <mj-raw>
+        <tr>
+            <td>
+                <div class="postNameAndTime">
```
Comment (AshishDhama) [inline templates/partials/sender_info_col.mjml:5]: Can't we put this class on the `td` tag?

**Review by AshishDhama — APPROVED** [review]
LGTM just few questions

**Review by matthewbirtch — APPROVED** [review]
Looks good to me!

File: templates/partials/sender_info_col.mjml (hunk 2 above)
Comment (harshilsharma63) [reply templates/partials/sender_info_col.mjml:5]: We can put it wherever we want.

File: app/notification_email_test.go [hunk 3]
```diff
@@ -339,6 +339,103 @@ func TestGetNotificationEmailBodyFullNotificationLocaleTime24Hour(t *testing.T)
...
+
+	storeMock := th.App.Srv().Store().(*mocks.Store)
+	teamStoreMock := mocks.TeamStore{}
+	teamStoreMock.On("GetByName", "testteam").Return(&model.Team{Name: "testteam"}, nil)
+	storeMock.On("Team").Return(&teamStoreMock)
+
+	body, err := th.App.getNotificationEmailBody(th.Context, recipient, post, channel, channelName, senderName, teamName, teamURL, emailNotificationContentsType, true, translateFunc, "user-avatar.png")
+	require.NoError(t, err)
+	require.Contains(t, body, "#FF0000")
+	require.Contains(t, body, "slack attachment 1 pretext")
```
[the remark this answers is not in the document]
Comment (harshilsharma63) [reply app/notification_email_test.go]: Done.

File: templates/partials/sender_info_col.mjml (hunk 2 above)
Comment (AshishDhama) [reply templates/partials/sender_info_col.mjml:5]: I guess we can remove the extra div, but this isn't blocking anything so up to you. Thanks.

**Pull request merged**